src/MementoContainer/Analysis/CollectionAnalyzer.cs

```diff
@@ -47,5 +62,25 @@ public IEnumerable<ICollectionData> GetCollections(object obj)
 
             return collections;
         }
+
+        /// <summary>
+        /// Tells if a given property has to be registered in the memento in case the class is decorated with MementoClassAttribute
+        /// </summary>
+        /// <param name="kv">Property info and its attributes</param>
+        /// <returns>True if the property must be registered in the memento</returns>
+        protected virtual bool PropertySelectorForMementoClass(KeyValuePair<PropertyInfo, IList<Attribute>> kv)
```

Comment:
Can we use an approach similar to what we did for `PropertyAnalyzer` for `CollectionAnalyzer` as well?